opengrok-tools/src/main/python/opengrok_tools/utils/log.py

```diff
@@ -122,7 +125,40 @@ def get_console_logger(name, level=logging.INFO, format='%(message)s'):
     logger = logging.getLogger(name)
     logger.setLevel(level)
     logger.propagate = False
+    logger.handlers = []
```

Comment:
The problem is that each project needs to log to its own log file. This is done via the `RotatingFileHandler`.

Reply:
aha, so you get the same instance of loggers but attache different files, seems fine

Reply:
I was thinking about making the logger tree wide so that each project would get its own logger that would be directly under the root logger, i.e. creating a logger hierarchy.

The logger used by a sub-module would look like  `opengrok_tools.<PROJECT_NAME>.foo.bar`. The `get_batch_logger()` is called with base class name, i.e. `opengrok_tools` and is non-propagating. Hence, log message emitted by `bar` would propagate up to the batch logger and stop there.

However, I am not sure what trick to use to inject the project part so that `logging.getLogger(__name__)` works seamlessly in sub-modules.

Reply:
A colleague pointed out that the `handlers` member is not starting with underscore so it can be probably be modified without too big risk.